From 50b64ce0a3046a2ebc847ef711f03d2b32e04348 Mon Sep 17 00:00:00 2001
From: Simon McVittie <smcv@debian.org>
Date: Tue, 7 Feb 2023 16:50:47 +0000
Subject: [PATCH 2/2] d/tests/installed-tests: Make the rules override work
 without polkitd-pkla

Installing policykit-1 is no longer guaranteed to provide support for
the legacy .pkla rules language.
---
 debian/tests/installed-tests                     | 2 +-
 debian/tests/overrides/99-allow-all-colord.pkla  | 4 ----
 debian/tests/overrides/99-allow-all-colord.rules | 9 +++++++++
 3 files changed, 10 insertions(+), 5 deletions(-)
 delete mode 100644 debian/tests/overrides/99-allow-all-colord.pkla
 create mode 100644 debian/tests/overrides/99-allow-all-colord.rules

diff --git a/debian/tests/installed-tests b/debian/tests/installed-tests
index 0ebf55ce..abf98144 100755
--- a/debian/tests/installed-tests
+++ b/debian/tests/installed-tests
@@ -3,7 +3,7 @@ set -eu
 
 # Override polkit checks for colord daemon. Normally this would allow
 # locally-logged-in users to do things, but our autopkgtest user isn't locally-logged-in.
-cp debian/tests/overrides/99-allow-all-colord.pkla /etc/polkit-1/localauthority/90-mandatory.d
+cp debian/tests/overrides/99-allow-all-colord.rules /etc/polkit-1/rules.d/
 
 mkdir -p /etc/systemd/system/colord.service.d/
 cp debian/tests/overrides/colord.service /etc/systemd/system/colord.service.d/10-add-dummy-sensor.conf
diff --git a/debian/tests/overrides/99-allow-all-colord.pkla b/debian/tests/overrides/99-allow-all-colord.pkla
deleted file mode 100644
index 18019af3..00000000
--- a/debian/tests/overrides/99-allow-all-colord.pkla
+++ /dev/null
@@ -1,4 +0,0 @@
-[Allow All for Tesnting]
-Identity=unix-user:*
-Action=org.freedesktop.color-manager.*
-ResultAny=yes
diff --git a/debian/tests/overrides/99-allow-all-colord.rules b/debian/tests/overrides/99-allow-all-colord.rules
new file mode 100644
index 00000000..b13fa6a3
--- /dev/null
+++ b/debian/tests/overrides/99-allow-all-colord.rules
@@ -0,0 +1,9 @@
+polkit.addRule(function(action, subject) {
+    if (action.id.indexOf("org.freedesktop.color-manager.") === 0) {
+        return polkit.Result.YES;
+    }
+
+    return polkit.Result.NOT_HANDLED;
+});
+
+// vim:set ft=javascript:
-- 
2.39.1

